examples_x035/usbpd_sink/usbpd.h

```diff
@@ -0,0 +1,921 @@
+	eUSBPD_APDO_RESERVED = 3,
+} USBPD_AugmentedPDO_e;
+
+typedef enum PACKED
+{
+	eUSBPD_PEAK_CURRENT_0 = 0, /* Peak Current equals IoC */
+
+	eUSBPD_PEAK_CURRENT_1 = 1, /* 150% IoC for 1ms @ 5% duty cycle
+	                              125% IoC for 2ms @ 10% duty cycle
+	                              110% IoC for 10ms @ 50% duty cycle */
+
+	eUSBPD_PEAK_CURRENT_2 = 2, /* 200% IoC for 1ms @ 5% duty cycle
+	                              150% IoC for 2ms @ 10% duty cycle
+	                              125% IoC for 10ms @ 50% duty cycle */
+
+	eUSBPD_PEAK_CURRENT_3 = 3, /* 200% IoC for 1ms @ 5% duty cycle
+	                              175% IoC for 2ms @ 10% duty cycle
+	                              150% IoC for 10ms @ 50% duty cycle */
+} USBPD_PeakCurrent_e;
+
+typedef struct
+{
+	uint32_t data : 28u; // PDO specific data based on PDO type
+	USBPD_AugmentedPDO_e AugmentedType : 2u; // shall be eUSBPD_APDO_SPR_PPS
+	USBPD_PowerDataObject_e PDOType : 2u; // shall be eUSBPD_PDO_AUGMENTED
+} USBPD_PDOHeader_t;
+
+typedef struct
+{
+	uint32_t MaxCurrentIn10mA : 10u;
+	uint32_t VoltageIn50mV : 10u;
+	USBPD_PeakCurrent_e PeakCurrent : 2u;
+	uint32_t Reserved_22bit : 1u;
+	uint32_t EPRModeCapable : 1u;
+	uint32_t UnchunkedExtendedMessage : 1u;
+	uint32_t DualRoleData : 1u;
+	uint32_t USBCommunicationsCapable : 1u;
+	uint32_t UnconstrainedPower : 1u;
+	uint32_t USBSuspendSupported : 1u;
+	uint32_t DualRolePower : 1u;
+	USBPD_PowerDataObject_e PDOType : 2u; // shall be eUSBPD_PDO_FIXED
+} USBPD_SourceFixedSupplyPDO_t;
+
+#define FIXED_SUPPLY_FMT                 \
+	"\nFixed Supply:\n"                  \
+	"\tMax Current: %d mA\n"             \
+	"\tVoltage: %d mV\n"                 \
+	"\tPeak Current: %d\n"               \
+	"\tEPR Mode Capable: %s\n"           \
+	"\tUnchunked Extended Message: %s\n" \
+	"\tDual Role Data: %s\n"             \
+	"\tUSB Communications Capable: %s\n" \
+	"\tUnconstrained Power: %s\n"        \
+	"\tUSB Suspend Supported: %s\n"      \
+	"\tDual Role Power: %s\n"
+
+#define FIXED_SUPPLY_FMT_ARGS( pdo )                                                                  \
+	( ( pdo )->FixedSupply.MaxCurrentIn10mA * 10 ), ( ( pdo )->FixedSupply.VoltageIn50mV * 50 ),      \
+		( ( pdo )->FixedSupply.PeakCurrent ), ( ( pdo )->FixedSupply.EPRModeCapable ? "Yes" : "No" ), \
+		( ( pdo )->FixedSupply.UnchunkedExtendedMessage ? "Yes" : "No" ),                             \
+		( ( pdo )->FixedSupply.DualRoleData ? "Yes" : "No" ),                                         \
+		( ( pdo )->FixedSupply.USBCommunicationsCapable ? "Yes" : "No" ),                             \
+		( ( pdo )->FixedSupply.UnconstrainedPower ? "Yes" : "No" ),                                   \
+		( ( pdo )->FixedSupply.USBSuspendSupported ? "Yes" : "No" ),                                  \
+		( ( pdo )->FixedSupply.DualRolePower ? "Yes" : "No" )
+
+typedef struct
+{
+	uint32_t MaxCurrentIn10mA : 10u;
+	uint32_t MinVoltageIn50mV : 10u;
```

Comment:
Does it matter that these bits line up and/or are aligned?  If so, please explain in a comment, if not, I recommend not defining the bit size.

Reply:
What do you mean? These are used to serialise and de-serialise the data. They are exactly as defined by the USB-PD spec.

Reply:
Please add the notes to where to find the information that, then.  

I am just saying that if you do use bit fields, there needs to be some explanation as to why they are bitfields.